### PR TITLE
Add APIs for ISV user managing policies belong to service template

### PR DESCRIPTION
**Pull request by baixinsui**
fixes https://github.com/eclipse-xpanse/xpanse/issues/1069

### CODE REVIEW

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServicePolicyManageApi.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+
+/**
+ * REST interface methods for service managing policies of the cloud service provider.
+ */
+@Slf4j
+@RestController
+@RequestMapping("/xpanse")
+@CrossOrigin
+@Secured({ROLE_ISV})
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServicePolicyManageApi.java:46]: just to keep it uniform with other controllers, we can add ADMIN_ROLE as well here.

File: modules/policy/src/main/java/org/eclipse/xpanse/modules/policy/policyman/PolicyManager.java [hunk 2]
```diff
@@ -44,16 +52,12 @@
 import org.springframework.web.client.RestClientException;
 
 /**
- * The service of policy-man.
+ * The service for managing policies.
  */
 @Slf4j
 @Component
 public class PolicyManager {
```
Comment (swaroopar) [inline modules/policy/src/main/java/org/eclipse/xpanse/modules/policy/policyman/PolicyManager.java:59]: this class has become too big. Would be good if we can split this into UserPolicyManager and ServicePolicyManager. We can move all common methods to another common class.

File: modules/policy/src/main/java/org/eclipse/xpanse/modules/policy/policyman/PolicyManager.java [hunk 3]
```diff
@@ -69,6 +73,167 @@ public class PolicyManager {
...
+        return conventToServicePolicy(existingEntity);
+    }
+
+    /**
+     * Delete the policy belonging to the registered service template.
+     *
+     * @param id the id of policy.
+     */
+    public void deleteServicePolicy(UUID id) {
+        ServicePolicyEntity existingEntity = getServicePolicyEntity(id, "delete");
```
Comment (swaroopar) [inline modules/policy/src/main/java/org/eclipse/xpanse/modules/policy/policyman/PolicyManager.java:167]: this variable is not used anywhere.

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServicePolicyManageApi.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+public class ServicePolicyManageApi {
+
+    @Resource
+    private PolicyManager policyManager;
+
+    @Resource
+    private IdentityProviderManager identityProviderManager;
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServicePolicyManageApi.java:53]: looks like this bean is not used in this class.

**Review by swaroopar — CHANGES_REQUESTED** [review]
Looks neat. Just very minor changes requested.

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServicePolicyManageApi.java (hunk 1 above)
Comment (baixinsui) [reply modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServicePolicyManageApi.java]: Updated as suggested.

File: modules/policy/src/main/java/org/eclipse/xpanse/modules/policy/policyman/PolicyManager.java (hunk 2 above)
Comment (baixinsui) [reply modules/policy/src/main/java/org/eclipse/xpanse/modules/policy/policyman/PolicyManager.java:33]: Updated as suggested.

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServicePolicyManageApi.java (hunk 4 above)
Comment (baixinsui) [reply modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServicePolicyManageApi.java]: Updated as suggested.

File: modules/policy/src/main/java/org/eclipse/xpanse/modules/policy/policyman/PolicyManager.java (hunk 3 above)
Comment (baixinsui) [reply modules/policy/src/main/java/org/eclipse/xpanse/modules/policy/policyman/PolicyManager.java]: Removed this variable.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**